pkg/operator/apis/preview.aro.openshift.io/v1alpha1/previewfeature_types.go

```diff
@@ -0,0 +1,87 @@
+package v1alpha1
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	operatorv1 "github.com/openshift/api/operator/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	SingletonPreviewFeatureName = "cluster"
+)
+
+// NSGFlowLogs defines NSG flow logs configuration
+type NSGFlowLogs struct {
+	// Enabled defines the behaviour of the reconciler:
+	// when true the controller will try to reach the desired configuration,
+	// when false it will try to disable the flow logs.
+	Enabled bool `json:"enabled"`
+
+	// Version defines version of NSG flow log.
+	// +kubebuilder:default:=2
+	// +kubebuilder:validation:Enum=1;2
+	Version int `json:"version,omitempty"`
+
+	// StorageAccountResourceID should be defined if BYO-Storage account is used.
+	// Must be in the same region of flow log.
+	StorageAccountResourceID string `json:"storageAccountResourceId,omitempty"`
+
+	// Retention period for data.
+	// +kubebuilder:default:="2160h"
+	Retention metav1.Duration `json:"retention,omitempty"`
+
+	// Required for TrafficAnalytics.
+	// Must be in the same region of flow log.
+	TrafficAnalyticsLogAnalyticsWorkspaceID string `json:"trafficAnalyticsLogAnalyticsWorkspaceId,omitempty"`
+
+	// Interval at which to conduct flow analytics. Values: 60m, 10m. Default: 60m.
+	// +kubebuilder:default:="60m"
+	// +kubebuilder:validation:Enum="60m";"10m"
+	TrafficAnalyticsInterval metav1.Duration `json:"trafficAnalyticsInterval,omitempty"`
+}
+
+// PreviewFeatureSpec defines the preview feature for ARO
+type PreviewFeatureSpec struct {
+	// NSGFlowLogs contains configuration for NSG flow logs.
+	// Omit the configuration if you don't want the controller
+	// to reconcile NSG flow logs.
+	NSGFlowLogs *NSGFlowLogs `json:"nsgFlowLogs,omitempty"`
+}
+
+// PreviewFeatureStatus defines the observed state of PreviewFeature
+type PreviewFeatureStatus struct {
+	OperatorVersion   string                         `json:"operatorVersion,omitempty"`
+	Conditions        []operatorv1.OperatorCondition `json:"conditions,omitempty"`
```

Comment:
This line and import in this file were the only chance I made manually in comparison to orignal PR #1670.

This channge caused auto-generated code to be a bit different, but that was expected.